protofsm/state_machine.go

```diff
@@ -228,17 +228,18 @@ func (s *StateMachine[Event, Env]) Stop() {
 // SendEvent sends a new event to the state machine.
 //
 // TODO(roasbeef): bool if processed?
-func (s *StateMachine[Event, Env]) SendEvent(event Event) {
-	s.sendEvent(event)
+func (s *StateMachine[Event, Env]) SendEvent(ctx context.Context, event Event) {
+	s.sendEvent(ctx, event)
 }
 
 // sendEvent sends a new event to the state machine.
-func (s *StateMachine[Event, Env]) sendEvent(event Event) {
+func (s *StateMachine[Event, Env]) sendEvent(ctx context.Context, event Event) {
```

Comment:
What was the motivation re the change in the prior commit? To only have the private method be what selects on the context?